CMakeLists.txt

```diff
@@ -1,42 +1,123 @@
-cmake_minimum_required(VERSION 3.20)
-set(CMAKE_OSX_ARCHITECTURES "x86_64")
-project(rack-hack VERSION 1.0 LANGUAGES C CXX)
+cmake_minimum_required(VERSION 3.16)
```

Comment:
Note: The rack-plugin-toolchain docker image currently only has CMake 3.16 installed.